checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakUtils.java

```diff
@@ -0,0 +1,118 @@
+package org.checkerframework.checker.resourceleak;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.checkerframework.checker.mustcall.MustCallChecker;
+import org.checkerframework.checker.mustcall.MustCallNoCreatesMustCallForChecker;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.rlccalledmethods.RLCCalledMethodsChecker;
+import org.checkerframework.framework.source.SourceChecker;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+
+/**
+ * Collection of static utility functions related to the various (sub-) checkers within the
+ * ResourceLeakChecker.
+ */
+public class ResourceLeakUtils {
+
+  /** Shouldn't be instantiated, pure utility class. */
+  public ResourceLeakUtils() {}
+
+  /** List of checker names associated with the Resource Leak Checker. */
+  public static List<String> rlcCheckers =
+      new ArrayList<>(
+          Arrays.asList(
+              ResourceLeakChecker.class.getCanonicalName(),
+              RLCCalledMethodsChecker.class.getCanonicalName(),
+              MustCallChecker.class.getCanonicalName(),
+              MustCallNoCreatesMustCallForChecker.class.getCanonicalName()));
+
+  /**
+   * Given a type factory part of the resource leak ecosystem, returns the {@link
+   * ResourceLeakChecker} in the checker hierarchy.
+   *
+   * @param referenceAtf the type factory to retrieve the {@link ResourceLeakChecker} from
+   * @return the {@link ResourceLeakChecker} in the checker hierarchy
+   */
+  public static @NonNull ResourceLeakChecker getResourceLeakChecker(
+      AnnotatedTypeFactory referenceAtf) {
+    if (referenceAtf == null) {
+      throw new IllegalArgumentException("Argument referenceAtf cannot be null");
+    } else {
+      return getResourceLeakChecker(referenceAtf.getChecker());
+    }
+  }
+
+  /**
+   * Given a checker part of the resource leak ecosystem, returns the {@link ResourceLeakChecker} in
+   * the checker hierarchy.
+   *
+   * @param referenceChecker the checker to retrieve the {@link ResourceLeakChecker} from
+   * @return the {@link ResourceLeakChecker} in the checker hierarchy
+   */
+  public static @NonNull ResourceLeakChecker getResourceLeakChecker(
+      SourceChecker referenceChecker) {
+    if (referenceChecker == null) {
```

Comment:
This is the not the implementation strategy that I would use for this method. In particular, I'd implement it like this:
```
if (referenceChecker instanceof ResourceLeakChecker) {
  return referenceChecker;
} else if (referenceChecker != null) {
  return getResourceLeakChecker(referenceChecker.getParent());
} else {
  throw ...
}
```

This has the advantage of avoiding hard-coding the string names of the various checkers. I don't see a disadvantage; is there one you're trying to avoid with the current implementation?